src/d3d12/d3d12_model_pool.hpp

```diff
@@ -46,7 +52,9 @@ namespace wr
 		void Evict() final;
 		void MakeResident() final;
 
-		void StageMeshes(d3d12::CommandList* cmd_list);
+		void StageMeshes();
```

Comment:
Return the generated command lists instead and mark it as [[nodiscard]]. The user shouldn't have to access the command list as a member variable. It would be less error prone if you just return a constant reference to it.